packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart

```diff
@@ -239,16 +239,49 @@ class FuchsiaRemoteConnection {
     _pollDartVms = false;
   }
 
+  /// Helper method for [getMainIsolatesByPattern].
+  ///
+  /// Called when either there are no Isolates that exist that match
+  /// `pattern`, or there are not yet any active Dart VM's on the system
+  /// (possible when the Isolate we're attempting to connect to is in the only
+  /// instance of the Dart VM and its service port has not yet opened).
+  Future<List<IsolateRef>> _waitForMainIsolatesByPattern([
+    Pattern pattern,
+    Duration timeout = _kIsolateFindTimeout,
+  ]) async {
+    final Completer<List<IsolateRef>> completer =
+        new Completer<List<IsolateRef>>();
+    _onDartVmEvent.listen(
+      (DartVmEvent event) async {
+        if (event.eventType == DartVmEventType.started) {
+          _log.fine('New VM found on port: ${event.servicePort}. Searching '
+              'for Isolate: $pattern');
+          final DartVm vmService = await _getDartVm(event.uri.port);
+          final List<IsolateRef> result =
+              await vmService.getMainIsolatesByPattern(pattern);
+          if (result.isNotEmpty) {
+            completer.complete(result);
```

Comment:
You may want to add a check here for `completer.isCompleted` (to handle the unexpected case of more than one event yielding a matching isolate ref)

Reply:
Sounds like a good idea.  Similar to the above, I wasn't sure how to handle something like just exiting this stream subscription after making it since I don't have an `onCancel` method implemented.

Does this mean this would run forever even after the `complete` function was called?  Right now the only thing that makes the listener function exit completely is when the `stop` function for `FuchsiaRemoteConnection` is called.

Reply:
Yeah, right now this could theoretically get invoked forever.  Ideally, maintain a reference to your stream subscription so you could cancel it, and you'd propagate that cancellation through to your connection.

In the meantime, you could just log a warning if you've already completed 😄

Reply:
sg